generator/dart/generated/google_cloud_functions_v2/lib/cloudfunctions.dart

```diff
@@ -327,33 +327,33 @@ class Function$ extends Message {
 }
 
 /// Describes the current state of the function.
-class Function$$State extends Enum {
+class Function$State extends Enum {
```

Comment:
Is the prefix *needed*?

Could this just be `State` without any prefix, without having a conflict in *this* package. Or even this library.
Otherwise `FunctionState` is reasonable too. 

Those are the two names I'd use if writing Dart code manually.

Reply:
> Is the prefix _needed_?

Maybe not now. Services evolve, if somebody adds a top-level `State` message then the prefix becomes needed and now you have a breaking change to deal with.

> Could this just be `State` without any prefix, without having a conflict in _this_ package. Or even this library. Otherwise `FunctionState` is reasonable too.
> 
> Those are the two names I'd use if writing Dart code manually.

OTOH, we need to create names that are stable under additions to the service definition. Without introducing breaking changes when the service makes a non-breaking change.

Reply:
I think the two goals are:

- having an ergonomic, idiomatic library
- not introducing unnecessary breaking api changes as a result of generation (not have the generator introduce a breaking change which isn't because of an upstream api breaking change)

Possibly another goal related to the 2nd item is that the delta in a generated library should be roughly proportional to an upstream API delta.